src/main/java/org/elasticsearch/common/xcontent/json/JsonXContentParser.java

```diff
@@ -88,7 +89,15 @@ public String text() throws IOException {
 
     @Override
     public BytesRef utf8Bytes() throws IOException {
-        return new BytesRef(CharBuffer.wrap(parser.getTextCharacters(), parser.getTextOffset(), parser.getTextLength()));
+        // Tentative workaround for https://github.com/elastic/elasticsearch/issues/8629
+        // TODO: Remove this when we upgrade jackson to 2.6.x.
+        char[] chars = parser.getTextCharacters();
+        if (chars == null) {
+            chars = EMPTY_CHARS;
```

Comment:
can we just `return new BytesRef()` in this case? I don't know if the text characters are null if we can really rely on text offset and length? Maybe a better check, not relying on null check is:

```
if (parser.getTextLength() == 0) {
    return new BytesRef();
}
```